# Exploring the Indian Startup Ecosystem: A Data Driven Analysis of Funding Trends and Industry Sectors

## Business Understanding
### Business Scenario
Your team is trying to venture into the Indian start-up
ecosystem. As the data expert of the team, you are to
investigate the ecosystem and propose the best course
of action.

*Analyze funding received by start-ups in India from
2018 to 2021.*
- Separate data for each year of funding will is
provided.
- In these datasets, you'll find the start-ups' details,
the funding amounts received, and the investors'
information.

### Business Objective
The aim of this project is to perform analysis on the Indian start-ups ecosystem and advice stakeholders on which venture to invest in to increase the potential of high profit/income.

## Hypothesis Testing
*Hypothesis* - The amount of funds a company receive depends on the sector a company finds itself
- Null Hypothesis(H_o) - The funds a company receive does not depend on the sector of investment
- Alternate Hypothesis(H_a) - The funds a company receive depends on the sector of investment

### Business Questions
- Which particular sector received the most funding over the time frame?
- The distribution of start ups in stages and the amount allocated each?
- What is the distribution of fundings based on locations?
- Which year had the most investors?
- Top 3 investor considerations in investing in start ups
- What was the impact of COVID-19 pandemic on startup funding in 2020?


# Data Loading and Exploration


## Data Dictionary

| Column Names|Description| Data Type|
|-------------|-----------|----------|
|Company/Brand|Name of company/start-up|Object|
|Founded|Year Start-up was founded|Int|
|Sector| Sector of service|Object|
|What it Does|Description about the company|Object|
|Founders| Founders of the company|Object|
|Investor| Investors|Object|
|Amount($)|Raised Founds|Float|
|Stage|Round of funding reached|Object|
|Headquarter|Location of company|Object|

## Data Understanding
Data Understanding phase drives the focus to identify, collect, and analyse
the data sets that can help accomplish the project goals. This
phase also has four tasks:


In [1]:
#import all necessary libraries

# data manipulation
import pandas as pd
import numpy as np

# data visualization libraries
import matplotlib.pyplot as plt
from plotly import express as px
import seaborn as sns

# statistical libraries
from scipy import stats
import statistics as stat

# database manipulation libraries
import pyodbc
from dotenv import dotenv_values

# hide warnings
import warnings
warnings.filterwarnings("ignore")
 

## Setup Database Connection

In [2]:
# load environment variables
environment_variables = dotenv_values(".env")

# load database configurations
database = environment_variables.get("DB_DATABASENAME")
server = environment_variables.get("DB_SERVER")
username = environment_variables.get("DB_USERNAME")
password = environment_variables.get("DB_PASSWORD")

# database connection string
connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}"



In [3]:
# create pyodbc connector
connection = pyodbc.connect(connection_string)


In [4]:
# Loading 2021 dataset from MS SQL server
query_2021 = " SELECT * FROM dbo.LP1_startup_funding2021"
df_2021 = pd.read_sql(query_2021,connection)
df_2021.head(2)

Company_Brand  Founded HeadQuarter      Sector  \
0  Unbox Robotics   2019.0   Bangalore  AI startup   
1          upGrad   2015.0      Mumbai      EdTech   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   

                                   Investor        Amount         Stage  
0               BEENEXT, Entrepreneur First    $1,200,000  Pre-series A  
1  Unilazer Ventures, IIFL Asset Management  $120,000,000          None

In [5]:
# Load 2020 dataset from MS SQL Server
query_2020 = "SELECT * FROM dbo.LP1_startup_funding2020"
df_2020 = pd.read_sql(query_2020,connection)
df_2020.head(2)

Company_Brand  Founded HeadQuarter    Sector  \
0    Aqgromalin   2019.0     Chennai  AgriTech   
1      Krayonnz   2019.0   Bangalore    EdTech   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   

                           Founders         Investor    Amount     Stage  \
0   Prasanna Manogaran, Bharani C L  Angel investors  200000.0      None   
1  Saurabh Dixit, Gurudutt Upadhyay  GSF Accelerator  100000.0  Pre-seed   

  column10  
0     None  
1     None

In [6]:
# load 2019 dataset
df_2019 = pd.read_csv("datasets/startup_funding2019.csv")
df_2019.head(2)

Company/Brand  Founded HeadQuarter     Sector  \
0  Bombay Shaving      NaN         NaN  Ecommerce   
1       Ruangguru   2014.0      Mumbai     Edtech   

                                        What it does  \
0         Provides a range of male grooming products   
1  A learning platform that provides topic-based ...   

                                Founders              Investor     Amount($)  \
0                     Shantanu Deshpande  Sixth Sense Ventures    $6,300,000   
1  Adamas Belva Syah Devara, Iman Usman.      General Atlantic  $150,000,000   

      Stage  
0       NaN  
1  Series C

In [7]:
# load 2018 dataset
df_2018 = pd.read_csv("datasets/startup_funding2018.csv")
df_2018.head(2)

Company Name                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   

  Round/Series       Amount                     Location  \
0         Seed       250000  Bangalore, Karnataka, India   
1         Seed  ₹40,000,000   Mumbai, Maharashtra, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...

# Data Assesment
After the data was loaded from the various sources, the data was visually assessed on Quality Issues.

## Data Quality Issues
We identified several data quality issues while exploring the datasets
1. Incosistent and missing columns:Some datasets were identified to have incosistent columns structures,with missing columns.This applies to the 2018 dataset.
2. Missing values and duplicates: There exists some null values and duplicates within the indivdual datasets.
3. Incosistent Values and currencies in the amount column: The amount column contains incosistent values and different currency types. 
4. Incosistent Values in the Sector,Location.,Industry Column: The following columns contain a lot of inconsistent values which needs to be looked at.



## Data Cleaning on the 2018 Dataset

In [8]:
# load dataset
df_2018.head(2)

Company Name                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   

  Round/Series       Amount                     Location  \
0         Seed       250000  Bangalore, Karnataka, India   
1         Seed  ₹40,000,000   Mumbai, Maharashtra, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...

In [9]:
#renaming columns
df_2018= df_2018.rename(columns=lambda x: x.lower().replace(' ', '_')) 
# add founded column to the dataset
df_2018["funded_year"] = 2018
df_2018.head(1)

company_name                                           industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   

  round/series  amount                     location  \
0         Seed  250000  Bangalore, Karnataka, India   

                                       about_company  funded_year  
0  TheCollegeFever is a hub for fun, fiesta and f...         2018

In [10]:
# check the shape of the 2018 dataset
df_2018.shape

(526, 7)

In [11]:
# perform descriptive statistics on the dataset
df_2018.describe(include="all").T

count unique  \
company_name     526    525   
industry         526    405   
round/series     526     21   
amount           526    198   
location         526     50   
about_company    526    524   
funded_year    526.0    NaN   

                                                             top freq    mean  \
company_name                                     TheCollegeFever    2     NaN   
industry                                                       —   30     NaN   
round/series                                                Seed  280     NaN   
amount                                                         —  148     NaN   
location                             Bangalore, Karnataka, India  102     NaN   
about_company  TheCollegeFever is a hub for fun, fiesta and f...    2     NaN   
funded_year                                                  NaN  NaN  2018.0   

               std     min     25%     50%     75%     max  
company_name   NaN     NaN     NaN     NaN     NaN     NaN  
industry       NaN     NaN     NaN     NaN     NaN     NaN  
round/series   NaN     NaN     NaN     NaN     NaN     NaN  
amount         NaN     NaN     NaN     NaN     NaN     NaN  
location       NaN     NaN     NaN     NaN     NaN     NaN  
about_company  NaN     NaN     NaN     NaN     NaN     NaN  
funded_year    0.0  2018.0  2018.0  2018.0  2018.0  2018.0

In [12]:
# check the info about the dataset
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   company_name   526 non-null    object
 1   industry       526 non-null    object
 2   round/series   526 non-null    object
 3   amount         526 non-null    object
 4   location       526 non-null    object
 5   about_company  526 non-null    object
 6   funded_year    526 non-null    int64 
dtypes: int64(1), object(6)
memory usage: 28.9+ KB


## Observations on the 2018 dataset
- The dataset consists of 526 rows and 6 columns
- All the data types of the columns are of the object type
- The amount column must be a numeric value
- There exist some duplicates in the dataset

### Course of Action
- Convert the amount column data type to float
- Drop duplicated rows

In [19]:
# check for null values
df_2018.isna().sum()

company_name     0
industry         0
round/series     0
amount           0
location         0
about_company    0
funded_year      0
dtype: int64

In [13]:
# check for duplicated values
df_2018.duplicated().sum()

1

In [14]:
# drop the duolicate value
df_2018.drop_duplicates(inplace=True)


## Cleaning of the Amount Column
Exchange Rate Between USD and Indian Rupee as of 2018

1USD = 68.4933INR

source: https://www.poundsterlinglive.com/history/USD-INR-2018

assumption: All the amount values without any currencies are assumed to be in USD

In [15]:
df_2018["amount"].unique()

array(['250000', '₹40,000,000', '₹65,000,000', '2000000', '—', '1600000',
       '₹16,000,000', '₹50,000,000', '₹100,000,000', '150000', '1100000',
       '₹500,000', '6000000', '650000', '₹35,000,000', '₹64,000,000',
       '₹20,000,000', '1000000', '5000000', '4000000', '₹30,000,000',
       '2800000', '1700000', '1300000', '₹5,000,000', '₹12,500,000',
       '₹15,000,000', '500000', '₹104,000,000', '₹45,000,000', '13400000',
       '₹25,000,000', '₹26,400,000', '₹8,000,000', '₹60,000', '9000000',
       '100000', '20000', '120000', '₹34,000,000', '₹342,000,000',
       '$143,145', '₹600,000,000', '$742,000,000', '₹1,000,000,000',
       '₹2,000,000,000', '$3,980,000', '$10,000', '₹100,000',
       '₹250,000,000', '$1,000,000,000', '$7,000,000', '$35,000,000',
       '₹550,000,000', '$28,500,000', '$2,000,000', '₹240,000,000',
       '₹120,000,000', '$2,400,000', '$30,000,000', '₹2,500,000,000',
       '$23,000,000', '$150,000', '$11,000,000', '₹44,000,000',
       '$3,240,000', '₹60

In [16]:
# create function to clean the amount column
exchange_rate = 68.4933
    
def clean_amount():
    # copy original amount columns
    amount_column = df_2018["amount"].copy().str.replace(",","") 
    # extract values in Rupees(₹)
    amount_in_rupee = amount_column[amount_column.str.startswith("₹")]
    # strip off the ₹ symbol
    amount_in_rupee = amount_in_rupee.str.lstrip("₹")
    # convert the amount in rupee to USD by the exchange rate of 68.4933
    amount_in_rupee = amount_in_rupee.apply(lambda x: float(x)/exchange_rate)
    # extract values in dollars($)
    amount_in_dollar = amount_column[amount_column.str.startswith("$")]
    # strip off the dollar symbol
    amount_in_dollar = amount_in_dollar.str.lstrip("$")
    # Replace the Unclean columns with the clean one
    amount_column.loc[amount_in_rupee.index] = amount_in_rupee
    amount_column.loc[amount_in_dollar.index] = amount_in_dollar
    
    # convert the clean column to numeric
    amount_column = pd.to_numeric(amount_column,errors="coerce")
    # update the amount column 
    df_2018["amount"] = amount_column




In [17]:
# call the clean_amount function
clean_amount()

In [19]:
# check info about the dataset after cleaning
df_2018.isnull().sum()


company_name       0
industry           0
round/series       0
amount           148
location           0
about_company      0
funded_year        0
dtype: int64

In [20]:
#renaming columns for simplicity
df_2018 = df_2018.rename(columns={"company_name":"company_brand", 
                                   "round/series":"stage",
                                   "about_company":"what_it_does"})

In [21]:
# perform a discriptive statistics on the new dataset
df_2018.describe(include="all").T

count unique                            top freq  \
company_brand    525    525                TheCollegeFever    1   
industry         525    405                              —   30   
stage            525     21                           Seed  279   
amount         377.0    NaN                            NaN  NaN   
location         525     50    Bangalore, Karnataka, India  101   
what_it_does     525    524  Algorithmic trading platform.    2   
funded_year    525.0    NaN                            NaN  NaN   

                          mean              std        min       25%  \
company_brand              NaN              NaN        NaN       NaN   
industry                   NaN              NaN        NaN       NaN   
stage                      NaN              NaN        NaN       NaN   
amount         17616765.176312  77972595.460917  875.99809  500000.0   
location                   NaN              NaN        NaN       NaN   
what_it_does               NaN              NaN        NaN       NaN   
funded_year             2018.0              0.0     2018.0    2018.0   

                     50%        75%           max  
company_brand        NaN        NaN           NaN  
industry             NaN        NaN           NaN  
stage                NaN        NaN           NaN  
amount         1300000.0  5000000.0  1000000000.0  
location             NaN        NaN           NaN  
what_it_does         NaN        NaN           NaN  
funded_year       2018.0     2018.0        2018.0

### Cleaning the stage column 

In [22]:
df_2018.stage.unique()

array(['Seed', 'Series A', 'Angel', 'Series B', 'Pre-Seed',
       'Private Equity', 'Venture - Series Unknown', 'Grant',
       'Debt Financing', 'Post-IPO Debt', 'Series H', 'Series C',
       'Series E', 'Corporate Round', 'Undisclosed',
       'https://docs.google.com/spreadsheets/d/1x9ziNeaz6auNChIHnMI8U6kS7knTr3byy_YBGfQaoUA/edit#gid=1861303593',
       'Series D', 'Secondary Market', 'Post-IPO Equity',
       'Non-equity Assistance', 'Funding Round'], dtype=object)

In [23]:
#Clean the Stage column categorize them
 
modified_stage = {
    'Seed': 'Seed funding',
    'Angel': 'Seed funding',
    'Venture - Series Unknown': 'Series A',
    'Debt Financing': 'Series A',
    'Grant': 'Seed funding',
    'Pre-Seed': 'Pre-Seed funding',
    'Funding Round': 'Pre-Seed funding',
    'Fresh funding': 'Pre-Seed funding',
    'Pre series A': 'Series A',
    'Post series A': 'Series A',
    'Seed fund': 'Seed funding',
    'Series B+': 'Series B',
    'Seed round': 'Seed funding',
    'Pre-seed' : 'Pre-Seed funding',
    'Seed A': 'Seed funding',
    'Pre-series A' : 'Series A',
    'Pre-series' : 'Pre-Seed funding',
    'Pre-series C' : 'Series C',
    'Pre-series B' : 'Series B',
    'Pre- series A' : 'Series A',
    'Pre-Series B' : 'Series B',
    'Series A-1' : 'Series A',
    'Seies A': 'Series A',
    'Seed Funding' : 'Seed funding',
    'Pre-Seed' : 'Pre-Seed funding',
    'Seed round': 'Seed funding' ,
    'Pre-seed Round': 'Pre-Seed funding',
    'Pre Series A' : 'Series A',
    'Pre seed Round': 'Pre-Seed funding',
    'Angel Round' :  'Seed funding',
    'Pre series A1': 'Series A',
    'Pre series A' : 'Series A',
    'Seed Round' : 'Seed funding',
    'Pre seed round' : 'Pre-Seed funding',
    'Pre series B' : 'Series B',
    'Pre series C' : 'Series B',
    'Seed Investment': 'Seed funding',
    'Seed+' : 'Seed funding',
    'Series A+' : 'Series A',
    'PE' : 'Private Equity',
    'Pre-series A1' : 'Series A',
    'Early seed' : 'Pre-Seed funding',
    'Series B2' : 'Series B',
    'Series E2' : 'Series E',
    'Series D1' : 'Series D',
    'Series B3' : 'Series B',
    'Series F1' : 'Series F',
    'Series A2' : 'Series A',
    'Series F2':'Series F',
    'Series C, D': 'Series C',
    'Bridge Round': 'Bridge',
    'https://docs.google.com/spreadsheets/d/1x9ziNeaz6auNChIHnMI8U6kS7knTr3byy_YBGfQaoUA/edit#gid=1861303593' : 'None',
    'ah! Ventures':'Series A', 
    'ITO Angel Network, LetsVenture': 'Seed funding',
    'JITO Angel Network, LetsVenture': 'Seed Funding', 
}

In [24]:
# Replacing the sector column values with the new sectors
df_2018["stage"]= df_2018["stage"].replace(modified_stage)
df_2018.stage.unique()

array(['Seed funding', 'Series A', 'Series B', 'Pre-Seed funding',
       'Private Equity', 'Post-IPO Debt', 'Series H', 'Series C',
       'Series E', 'Corporate Round', 'Undisclosed', 'None', 'Series D',
       'Secondary Market', 'Post-IPO Equity', 'Non-equity Assistance'],
      dtype=object)

## Observations
- There is a vast difference between the mean and the median of the amount column which suggests outliers present in the dataset
- There exists some null values still present in the dataset 

## Course Of Actions
Fillna Values in the amount column:Impute the NaN values in the amount column with the median because of the outliers

In [25]:
# Function to fill NaN values in the amount column with the average based on stage of company
def impute_amount_column(df,filter_name,fill_value):
    unique_values = df[filter_name].unique()
    for val, avg_amount in zip(unique_values,df.groupby(filter_name)[fill_value].transform('median')):
        df.loc[df[filter_name] == val, fill_value] = df.loc[df[filter_name] == val, fill_value].fillna(avg_amount)
    return df
     


In [26]:
# call the impute_amount_column function
df_2018 = impute_amount_column(df_2018,"stage","amount")

In [27]:
# Split the Industry Column
industry_split = df_2018["industry"].str.split(",", expand=True)
industry_split.head()

0                    1           2             3   \
0       Brand Marketing      Event Promotion   Marketing   Sponsorship   
1           Agriculture              Farming        None          None   
2                Credit   Financial Services     Lending   Marketplace   
3    Financial Services              FinTech        None          None   
4  E-Commerce Platforms               Retail        SaaS          None   

           4     5     6     7     8     9     10    11  
0   Ticketing  None  None  None  None  None  None  None  
1        None  None  None  None  None  None  None  None  
2        None  None  None  None  None  None  None  None  
3        None  None  None  None  None  None  None  None  
4        None  None  None  None  None  None  None  None

In [28]:
# concatenate industry_split with the original df_2018
df_2018 = pd.concat([df_2018,industry_split[0]], ignore_index=False, axis=1)
df_2018.head(1)

company_brand                                           industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   

          stage    amount                     location  \
0  Seed funding  250000.0  Bangalore, Karnataka, India   

                                        what_it_does  funded_year  \
0  TheCollegeFever is a hub for fun, fiesta and f...         2018   

                 0  
0  Brand Marketing

In [29]:
# Rename the column 0 to sector
df_2018 = df_2018.rename(columns={0:"sector"})
df_2018.head()

company_brand                                           industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

          stage        amount                          location  \
0  Seed funding  2.500000e+05       Bangalore, Karnataka, India   
1  Seed funding  5.839987e+05        Mumbai, Maharashtra, India   
2      Series A  9.489979e+05           Gurgaon, Haryana, India   
3  Seed funding  2.000000e+06       Noida, Uttar Pradesh, India   
4  Seed funding  5.669994e+05  Hyderabad, Andhra Pradesh, India   

                                        what_it_does  funded_year  \
0  TheCollegeFever is a hub for fun, fiesta and f...         2018   
1  A startup which aggregates milk from dairy far...         2018   
2          Leading Online Loans Marketplace in India         2018   
3  PayMe India is an innovative FinTech organizat...         2018   
4  Eunimart is a one stop solution for merchants ...         2018   

                 sector  
0       Brand Marketing  
1           Agriculture  
2                Credit  
3    Financial Services  
4  E-Commerce Platforms

In [31]:
# Rename the column 0 to sector
df_2018 = df_2018.drop(columns={"industry"})

## Restructuring the Sector Column into Key Sectors of the Economy
source: https://www.businessinsider.in/business/startups/news/top-10-industries-for-new-startups-in-india-as-per-hurun-list/articleshow/105651758.cms

In [33]:
# Check for unique values in the sector column 

df_2018["sector"].sort_values().unique()

array(['3D Printing', 'Accounting', 'Advertising', 'Aerospace', 'AgTech',
       'Agriculture', 'Air Transportation', 'Alternative Medicine',
       'Analytics', 'Android', 'Apps', 'Artificial Intelligence', 'Audio',
       'Automotive', 'Autonomous Vehicles', 'B2B', 'Banking',
       'Basketball', 'Battery', 'Beauty', 'Big Data', 'Biopharma',
       'Biotechnology', 'Blockchain', 'Brand Marketing', 'Broadcasting',
       'Business Development', 'Business Intelligence', 'Business Travel',
       'Career Planning', 'Catering', 'Child Care', 'Children',
       'Classifieds', 'Clean Energy', 'CleanTech', 'Cloud Computing',
       'Cloud Infrastructure', 'Collaboration', 'Commercial',
       'Commercial Real Estate', 'Communities', 'Computer', 'Consulting',
       'Consumer', 'Consumer Applications', 'Consumer Electronics',
       'Consumer Goods', 'Consumer Lending', 'Continuing Education',
       'Cooking', 'Cosmetics', 'Creative Agency', 'Credit',
       'Credit Cards', 'Crowdfunding', 

From research, there are 12 sectors in India where all these values in the sector column of the dataset can be generalised into. They are therefore mapped onto the 12 sectors which include:

- IT & Technology               
- Financial Services             
- Healthcare                     
- Consumer Goods                 
- Business Services              
- Media & Entertainment         
- Education                      
- Others                         
- Manufacturing                  
- Retail                         
- Hospitality                    
- Transportation & Logistics     
- Sports                         
- Agriculture                    
- Real Estate      

In [34]:
sector_mapping = {
    '3D Printing': 'IT & Technology',
    'Accounting': 'Business Services',
    'Advertising': 'Media & Entertainment',
    'Aerospace': 'Manufacturing',
    'AgTech': 'Agriculture',
    'Agriculture': 'Agriculture',
    'Air Transportation': 'Transportation & Logistics',
    'Alternative Medicine': 'Healthcare',
    'Analytics': 'IT & Technology',
    'Android': 'IT & Technology',
    'Apps': 'IT & Technology',
    'Artificial Intelligence': 'IT & Technology',
    'Audio': 'Media & Entertainment',
    'Automotive': 'Manufacturing',
    'Autonomous Vehicles': 'Manufacturing',
    'B2B': 'Business Services',
    'Banking': 'Financial Services',
    'Basketball': 'Sports',
    'Battery': 'IT & Technology',
    'Beauty': 'Consumer Goods',
    'Big Data': 'IT & Technology',
    'Biopharma': 'Healthcare',
    'Biotechnology': 'Healthcare',
    'Blockchain': 'IT & Technology',
    'Brand Marketing': 'Media & Entertainment',
    'Broadcasting': 'Media & Entertainment',
    'Business Development': 'Business Services',
    'Business Intelligence': 'Business Services',
    'Business Travel': 'Business Services',
    'Career Planning': 'Business Services',
    'Catering': 'Hospitality',
    'Child Care': 'Business Services',
    'Children': 'Consumer Goods',
    'Classifieds': 'Media & Entertainment',
    'Clean Energy': 'IT & Technology',
    'CleanTech': 'IT & Technology',
    'Cloud Computing': 'IT & Technology',
    'Cloud Infrastructure': 'IT & Technology',
    'Collaboration': 'IT & Technology',
    'Commercial': 'Real Estate',
    'Commercial Real Estate': 'Real Estate',
    'Communities': 'IT & Technology',
    'Computer': 'IT & Technology',
    'Consulting': 'Business Services',
    'Consumer': 'Consumer Goods',
    'Consumer Applications': 'IT & Technology',
    'Consumer Electronics': 'Consumer Goods',
    'Consumer Goods': 'Consumer Goods',
    'Consumer Lending': 'Financial Services',
    'Continuing Education': 'Education',
    'Cooking': 'Hospitality',
    'Cosmetics': 'Consumer Goods',
    'Creative Agency': 'Business Services',
    'Credit': 'Financial Services',
    'Credit Cards': 'Financial Services',
    'Crowdfunding': 'Financial Services',
    'Crowdsourcing': 'Business Services',
    'Cryptocurrency': 'Financial Services',
    'Customer Service': 'Business Services',
    'Dating': 'Media & Entertainment',
    'Delivery': 'Transportation & Logistics',
    'Delivery Service': 'Transportation & Logistics',
    'Dental': 'Healthcare',
    'Dietary Supplements': 'Healthcare',
    'Digital Entertainment': 'Media & Entertainment',
    'Digital Marketing': 'Media & Entertainment',
    'Digital Media': 'Media & Entertainment',
    'E-Commerce': 'Retail',
    'E-Commerce Platforms': 'Retail',
    'E-Learning': 'Education',
    'EdTech': 'Education',
    'Education': 'Education',
    'Electric Vehicle': 'Transportation & Logistics',
    'Embedded Systems': 'IT & Technology',
    'Energy': 'IT & Technology',
    'Enterprise Resource Planning (ERP)': 'IT & Technology',
    'Enterprise Software': 'IT & Technology',
    'Environmental Consulting': 'Business Services',
    'Events': 'Hospitality',
    'Eyewear': 'Consumer Goods',
    'Facilities Support Services': 'Business Services',
    'Fantasy Sports': 'Sports',
    'Farming': 'Agriculture',
    'Fashion': 'Consumer Goods',
    'File Sharing': 'IT & Technology',
    'FinTech': 'Financial Services',
    'Finance': 'Financial Services',
    'Financial Services': 'Financial Services',
    'Fitness': 'Sports',
    'Food Delivery': 'Hospitality',
    'Food Processing': 'Consumer Goods',
    'Food and Beverage': 'Consumer Goods',
    'Fraud Detection': 'Financial Services',
    'Funding Platform': 'Financial Services',
    'Gaming': 'Media & Entertainment',
    'Government': 'Business Services',
    'Health Care': 'Healthcare',
    'Health Diagnostics': 'Healthcare',
    'Health Insurance': 'Financial Services',
    'Home Decor': 'Consumer Goods',
    'Hospital': 'Healthcare',
    'Hospitality': 'Hospitality',
    'Human Resources': 'Business Services',
    'Industrial': 'Manufacturing',
    'Industrial Automation': 'Manufacturing',
    'Information Services': 'IT & Technology',
    'Information Technology': 'IT & Technology',
    'Insurance': 'Financial Services',
    'Internet': 'IT & Technology',
    'Internet of Things': 'IT & Technology',
    'Last Mile Transportation': 'Transportation & Logistics',
    'Logistics': 'Transportation & Logistics',
    'Manufacturing': 'Manufacturing',
    'Market Research': 'Business Services',
    'Marketing': 'Media & Entertainment',
    'Marketplace': 'Retail',
    'Media and Entertainment': 'Media & Entertainment',
    'Medical': 'Healthcare',
    'Medical Device': 'Healthcare',
    'Mobile': 'IT & Technology',
    'Mobile Payments': 'Financial Services',
    'Music': 'Media & Entertainment',
    'Music Streaming': 'Media & Entertainment',
    'Nanotechnology': 'IT & Technology',
    'News': 'Media & Entertainment',
    'Online Games': 'Media & Entertainment',
    'Online Portals': 'Media & Entertainment',
    'Packaging Services': 'Manufacturing',
    'Reading Apps': 'Media & Entertainment',
    'Renewable Energy': 'IT & Technology',
    'Rental': 'Real Estate',
    'Retail': 'Retail',
    'Search Engine': 'IT & Technology',
    'Smart Cities': 'IT & Technology',
    'Social Media': 'Media & Entertainment',
    'Software': 'IT & Technology',
    'Sports': 'Sports',
    'Tourism': 'Hospitality',
    'Trading Platform': 'Financial Services',
    'Training': 'Education',
    'Transportation': 'Transportation & Logistics',
    'Travel': 'Hospitality',
    'Veterinary': 'Healthcare',
    'Wealth Management': 'Financial Services',
    'Wedding': 'Hospitality',
    'Wellness': 'Healthcare',
    'eSports': 'Sports',
    '—': 'Others'
}


In [36]:
# Replacing the sector column values with the new sectors
df_2018["sector"]= df_2018["sector"].replace(sector_mapping)
df_2018.head(2)

company_brand         stage         amount                     location  \
0  TheCollegeFever  Seed funding  250000.000000  Bangalore, Karnataka, India   
1  Happy Cow Dairy  Seed funding  583998.726883   Mumbai, Maharashtra, India   

                                        what_it_does  funded_year  \
0  TheCollegeFever is a hub for fun, fiesta and f...         2018   
1  A startup which aggregates milk from dairy far...         2018   

                  sector  
0  Media & Entertainment  
1            Agriculture

In [37]:
df_2018["sector"].value_counts()

sector
IT & Technology               116
Financial Services             77
Healthcare                     45
Consumer Goods                 44
Business Services              36
Media & Entertainment          32
Education                      31
Others                         30
Manufacturing                  29
Retail                         26
Hospitality                    19
Transportation & Logistics     14
Sports                         12
Agriculture                    10
Real Estate                     4
Name: count, dtype: int64

In [38]:
others_cat = df_2018[df_2018["sector"] == "Others" ]
others_cat

company_brand             stage  \
58                        MissMalini Entertainment      Seed funding   
105                               Jagaran Microfin          Series A   
121                                         FLEECA      Seed funding   
146                                      WheelsEMI          Series B   
153                                    Fric Bergen          Series A   
174                                       Deftouch      Seed funding   
181                                    Corefactors      Seed funding   
210                                Cell Propulsion      Seed funding   
230                                       Flathalt      Seed funding   
235                                          dishq      Seed funding   
238                                          Trell      Seed funding   
242                                 HousingMan.com      Seed funding   
243                       Steradian Semiconductors      Seed funding   
247                                   SaffronStays      Seed funding   
251                           Inner Being Wellness      Seed funding   
257                                       MySEODoc      Seed funding   
258       ENLYFT DIGITAL SOLUTIONS PRIVATE LIMITED      Seed funding   
259                                     Scale Labs      Seed funding   
294                                       Roadcast      Seed funding   
325                                         Toffee      Seed funding   
326                                     ORO Wealth          Series A   
346                                        Finwego      Seed funding   
398                                           Cred      Seed funding   
457                                          Origo          Series A   
472                                      Sequretek          Series A   
473               Avenues Payments India Pvt. Ltd.   Corporate Round   
476  Planet11 eCommerce Solutions India (Avenue11)   Corporate Round   
478                                 Iba Halal Care          Series A   
480                                         Togedr          Series A   
510                                       Scholify  Pre-Seed funding   

           amount                          location  \
58   1.518397e+06        Mumbai, Maharashtra, India   
105  8.029982e+06       Kolkata, West Bengal, India   
121  5.669994e+05          Jaipur, Rajasthan, India   
146  1.400000e+07          Pune, Maharashtra, India   
153  5.669994e+05           Alwar, Rajasthan, India   
174  5.669994e+05       Bangalore, Karnataka, India   
181  5.669994e+05       Bangalore, Karnataka, India   
210  1.021998e+05       Bangalore, Karnataka, India   
230  5.000000e+04           Gurgaon, Haryana, India   
235  4.000000e+05       Bengaluru, Karnataka, India   
238  1.250000e+06  Bangalore City, Karnataka, India   
242  1.503797e+06       Bengaluru, Karnataka, India   
243  5.669994e+05       Bangalore, Karnataka, India   
247  2.000000e+06        Mumbai, Maharashtra, India   
251  5.109989e+05  Hyderabad, Andhra Pradesh, India   
257  5.669994e+05               Delhi, Delhi, India   
258  2.919994e+03          Pune, Maharashtra, India   
259  1.000000e+06               Delhi, Delhi, India   
294  2.500000e+05           New Delhi, Delhi, India   
325  1.500000e+06           Gurgaon, Haryana, India   
326  1.600000e+06        Mumbai, Maharashtra, India   
346  5.669994e+05        Chennai, Tamil Nadu, India   
398  5.669994e+05       Bengaluru, Karnataka, India   
457  1.167997e+07           Gurgaon, Haryana, India   
472  3.700000e+06        Mumbai, Maharashtra, India   
473  1.300000e+06        Mumbai, Maharashtra, India   
476  3.795992e+06       Bengaluru, Karnataka, India   
478  2.189995e+06         Ahmedabad, Gujarat, India   
480  5.669994e+05           Gurgaon, Haryana, India   
510  5.669994e+05       Bengaluru, Karnataka, India   

                                          what_it_does  funded_year  sector  
58   MissMalini Ente

## Recategorizing the Others Values into their appropriate sectors

In [39]:
# view the descriptions of the companies with the others categories
others_cat["what_it_does"]

58     MissMalini Entertainment is a multi-platform n...
105    Jagaran Microfin is a Microfinance institution...
121              FLEECA is a Tyre Care Provider company.
146    WheelsEMI is the brand name of NBFC, WheelsEMI...
153    Fric Bergen is a leader in the specialty food ...
174    Deftouch is a mobile game development company ...
181    Corefactors is a leading campaign management, ...
210    Cell Propulsion is an electric mobility startu...
230                      FInd your Customized Home here.
235    dishq leverages food science and machine learn...
238    Trell is a location based network which helps ...
242          New Apartments, Flats for Sale in Bangalore
243    It is a fabless semiconductor company focused ...
247    SaffronStays connects travellers to India's In...
251    Inner Being Wellness manufactures beauty, well...
257    SEO, PPC, Search Engine Marketing, Social Medi...
258                             Digital Marketing Agency
259    Scale Labs is a cross bo

In [40]:
# create dictionary to contain the keywords in the what_it_does column
keywords = {
    'entertainment': 'Media & Entertainment',
    'microfinance': 'Financial Services',
    'tyre care': 'Consumer Goods',
    'nbfc': 'Financial Services',
    'specialty food': 'Consumer Goods',
    'mobile game development': 'IT & Technology',
    'campaign management': 'Business Services',
    'electric mobility startup': 'Transportation & Logistics',
    'food science': 'Consumer Goods',
    'machine learning': 'IT & Technology',
    'location based network': 'IT & Technology',
    'real estate': 'Real Estate',
    'semiconductor company': 'IT & Technology',
    'travellers accommodation': 'Hospitality',
    'beauty, wellness': 'Wellness',
    'search engine marketing': 'IT & Technology',
    'digital marketing agency': 'Business Services',
    'cross border e-commerce solutions': 'Business Services',
    'wealth management platform': 'Financial Services',
    'micro-event & contextual marketing': 'Business Services',
    'partners with small and medium businesses': 'Financial Services',
    'celebrate and reward': 'Financial Services',
    'post-harvest management': 'Business Services',
    'cyber security': 'IT & Technology',
    'cosmetics brand': 'Consumer Goods',
    'activity discovery & booking platform': 'Hospitality',
    'edutech': 'Education'
}


In [41]:

# Function to assign sector based on description
def assign_sector(what_it_does):
    for keyword,sector in keywords.items():
        if keyword in what_it_does.lower():
            return sector
    return "Others" #Keep Others if no keyword matches


In [42]:
# Update sectors for entries currently labeled as "Others"
df_2018.loc[df_2018['sector'] == 'Others', 'sector'] = df_2018.loc[df_2018['sector'] == 'Others', 'what_it_does'].apply(assign_sector)


In [43]:
# confirm changes
df_2018["sector"].value_counts()

sector
IT & Technology               121
Financial Services             82
Consumer Goods                 48
Healthcare                     45
Business Services              40
Media & Entertainment          33
Education                      32
Manufacturing                  29
Retail                         26
Hospitality                    20
Transportation & Logistics     15
Sports                         12
Agriculture                    10
Others                          7
Real Estate                     4
Wellness                        1
Name: count, dtype: int64

## Cleaning the Location Column

### Observations From The Column
- The format of the data found in the column mostly are city,state,country format
### Tasks to Performed
- Split the Column values by comma
- Drop the state and the country columns
- Retain the city column and rename it headquarter to be in-line with the other three datasets

In [44]:
# look for unique values in the location column
df_2018["location"].unique()

array(['Bangalore, Karnataka, India', 'Mumbai, Maharashtra, India',
       'Gurgaon, Haryana, India', 'Noida, Uttar Pradesh, India',
       'Hyderabad, Andhra Pradesh, India', 'Bengaluru, Karnataka, India',
       'Kalkaji, Delhi, India', 'Delhi, Delhi, India', 'India, Asia',
       'Hubli, Karnataka, India', 'New Delhi, Delhi, India',
       'Chennai, Tamil Nadu, India', 'Mohali, Punjab, India',
       'Kolkata, West Bengal, India', 'Pune, Maharashtra, India',
       'Jodhpur, Rajasthan, India', 'Kanpur, Uttar Pradesh, India',
       'Ahmedabad, Gujarat, India', 'Azadpur, Delhi, India',
       'Haryana, Haryana, India', 'Cochin, Kerala, India',
       'Faridabad, Haryana, India', 'Jaipur, Rajasthan, India',
       'Kota, Rajasthan, India', 'Anand, Gujarat, India',
       'Bangalore City, Karnataka, India', 'Belgaum, Karnataka, India',
       'Thane, Maharashtra, India', 'Margão, Goa, India',
       'Indore, Madhya Pradesh, India', 'Alwar, Rajasthan, India',
       'Kannur, Kerala, Ind

In [45]:
# Split the location Column
df_2018[["headquarter", "state","country"]] = df_2018["location"].str.split(",", n=2, expand=True)
df_2018.head(1)

company_brand         stage    amount                     location  \
0  TheCollegeFever  Seed funding  250000.0  Bangalore, Karnataka, India   

                                        what_it_does  funded_year  \
0  TheCollegeFever is a hub for fun, fiesta and f...         2018   

                  sector headquarter       state country  
0  Media & Entertainment   Bangalore   Karnataka   India

In [53]:
# check for the unique states in the dataset
df_2018["state"].unique()

KeyError: 'state'

In [50]:
# drop the state column 
df_2018.drop(columns={"state"},inplace=True)

KeyError: "['state'] not found in axis"

In [51]:
# drop the country column 
df_2018.drop(columns={"country"},inplace=True)

In [52]:
df_2018['headquarter'].unique()

array(['Bangalore', 'Mumbai', 'Gurgaon', 'Noida', 'Hyderabad',
       'Bengaluru', 'Kalkaji', 'Delhi', 'India', 'Hubli', 'New Delhi',
       'Chennai', 'Mohali', 'Kolkata', 'Pune', 'Jodhpur', 'Kanpur',
       'Ahmedabad', 'Azadpur', 'Haryana', 'Cochin', 'Faridabad', 'Jaipur',
       'Kota', 'Anand', 'Bangalore City', 'Belgaum', 'Thane', 'Margão',
       'Indore', 'Alwar', 'Kannur', 'Trivandrum', 'Ernakulam',
       'Kormangala', 'Uttar Pradesh', 'Andheri', 'Mylapore', 'Ghaziabad',
       'Kochi', 'Powai', 'Guntur', 'Kalpakkam', 'Bhopal', 'Coimbatore',
       'Worli', 'Alleppey', 'Chandigarh', 'Guindy', 'Lucknow'],
      dtype=object)

In [54]:
# Modified location mapping
modified_location ={
    'Bengaluru': 'Bangalore',
    'Delhi': 'New Delhi',
    'Bangalore City' : 'Bangalore',
    #'Gurgaon' : 'Gurugram',
    'Haryana': 'Chandigarh',
    'Uttar Pradesh' : 'Noida',
    'Uttar pradesh' : 'Noida',
    'Tamil Nadu': 'Chennai',
    'Faridabad, Haryana': 'Faridabad',
    'Computer Games': 'Undisclosed',
    'Telangana': 'Hyderabad',  # 2019
    'Dhingsara, Haryana': 'Chandigarh',
    'The nilgiris': 'Chennai',
    'Santra': 'Bhopal',
    'India': 'New delhi',
    'Gurugram\\t#ref!': 'Gurgaon',
    '"Gurugram\t#ref!"': 'Gurgaon',
    'Banglore': 'Bangalore',
    'Singapore': 'Overseas',
    'Frisco, Texas, United States': 'Overseas',
    'California': 'Overseas',
    'New York, United States': 'Overseas',
    'San Francisco, California, United States': 'Overseas',
    'San Francisco, United States': 'Overseas',
    'San Ramon, California': 'Overseas',
    'Paris, Ile-de-France, France': 'Overseas',
    'Plano, Texas, United States': 'Overseas',
    'Sydney': 'Overseas',
    'San Francisco Bay Area, Silicon Valley, West Coast': 'Overseas',
    'Bangaldesh': 'Overseas',
    'London, England, United Kingdom': 'Overseas',
    'Sydney, New South Wales, Australia': 'Overseas',
    'Milano, Lombardia, Italy': 'Overseas',
    'Palmwoods, Queensland, Australia': 'Overseas',
    'France': 'Overseas',
    'San Francisco Bay Area, West Coast, Western US': 'Overseas',
    'Tangerang, Jawa Barat, Indonesia': 'Overseas',
    'Berlin': 'Overseas',
    'Seattle': 'Overseas',
    'Riyadh': 'Overseas',
    'Seoul': 'Overseas',
    'New York': 'Overseas',
    'Bangkok': 'Overseas',
    'Newcastle Upon Tyne, Newcastle upon Tyne, United Kingdom': 'Overseas',
    'Shanghai, China': 'Overseas',
    'Jiaxing, Zhejiang, China': 'Overseas',
    'Irvine, California, United States': 'Overseas',
    'San Franciscao' : 'Overseas',
    'San Francisco': 'Overseas',
    'Jaipur, Rajastan': 'Jaipur',
    'Rajastan': "Jaipur",
    'Tumkur, Karnataka': 'Tumkur',
    'Samastipur, Bihar': 'Samastipur',
    #'None' : 'Undisclosed',
    'Tumkur, Karnataka' : 'Tumkur',
    'Information Technology & Services': 'Undisclosed',
    'Tirunelveli, Tamilnadu': 'Tirunelveli',
    'Trivandrum, Kerala, India': 'Trivandrum',
    'Manchester': 'Overseas',
    'Mountain View, CA' : 'Overseas',
    'London' : 'Overseas',
    'The Nilgiris' :'Chennai',
    'Beijing' : 'Overseas',
    'Small Towns, Andhra Pradesh': 'Amaravati',
    'Goa' : 'Panaji',
    'West Bengal' : 'Kolkata',
    'Telugana': 'Telangana',
    'Orissia': 'Odisha',
    'Jharkhand': 'Ranchi',
    'Thiruvananthapuram': 'Trivandrum',
    'Gurugram\t#REF!': 'Gurgaon'
   
 
 
}

In [55]:
df_2018.replace({'headquarter': modified_location}, inplace=True)

In [56]:
df_2018["headquarter"] = df_2018["headquarter"].replace("India","New Delhi")

## Data Cleaning on the 2019 Dataset

In [57]:
df_2019= df_2019.rename(columns=lambda x: x.lower().replace(' ', '_')) 
#renaming columns
df_2019 = df_2019.rename(columns={"company/brand":"company_brand", "amount($)":"amount"})
#adding year column
df_2019["funded_year"] = 2019

In [58]:
df_2019.drop_duplicates(inplace=True)

In [65]:
df_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   company_brand  89 non-null     object 
 1   founded        60 non-null     float64
 2   headquarter    70 non-null     object 
 3   sector         84 non-null     object 
 4   what_it_does   89 non-null     object 
 5   founders       86 non-null     object 
 6   investor       89 non-null     object 
 7   amount         77 non-null     float64
 8   stage          43 non-null     object 
 9   funded_year    89 non-null     int64  
dtypes: float64(2), int64(1), object(7)
memory usage: 7.1+ KB


### Cleaning the amount column

In [59]:
df_2019.amount.unique()

array(['$6,300,000', '$150,000,000', '$28,000,000', '$30,000,000',
       '$6,000,000', 'Undisclosed', '$1,000,000', '$20,000,000',
       '$275,000,000', '$22,000,000', '$5,000,000', '$140,500',
       '$540,000,000', '$15,000,000', '$182,700', '$12,000,000',
       '$11,000,000', '$15,500,000', '$1,500,000', '$5,500,000',
       '$2,500,000', '$140,000', '$230,000,000', '$49,400,000',
       '$32,000,000', '$26,000,000', '$150,000', '$400,000', '$2,000,000',
       '$100,000,000', '$8,000,000', '$100,000', '$50,000,000',
       '$120,000,000', '$4,000,000', '$6,800,000', '$36,000,000',
       '$5,700,000', '$25,000,000', '$600,000', '$70,000,000',
       '$60,000,000', '$220,000', '$2,800,000', '$2,100,000',
       '$7,000,000', '$311,000,000', '$4,800,000', '$693,000,000',
       '$33,000,000'], dtype=object)

In [60]:
df_2019.loc[df_2019.amount== 'Undisclosed']

company_brand  founded headquarter            sector  \
5                       FlytBase      NaN        Pune        Technology   
6                          Finly      NaN   Bangalore              SaaS   
10                    Cub McPaws   2010.0      Mumbai   E-commerce & AR   
14                   Open Secret      NaN         NaN         Food tech   
19  Azah Personal Care Pvt. Ltd.   2018.0    Gurugram            Health   
23            DROR Labs Pvt. Ltd   2018.0       Delhi       Safety tech   
32                      Pumpkart   2014.0  Chandigarh     E-marketplace   
45                        Afinoz      NaN       Noida           Fintech   
54                     Ninjacart   2015.0         NaN  B2B Supply Chain   
55                   Binca Games   2014.0      Mumbai             Games   
65                  Cubical Labs   2013.0         NaN               IoT   
75                      Sportido   2015.0       Noida            Sports   

                                         what_it_does  \
5                         A drone automation platform   
6   It builds software products that makes work si...   
10  A B2C  brand that focusses on premium and comf...   
14           It produces and sells top quality snacks   
19  Aims to solve some problems in the feminine hy...   
23  It uses technology to create a trust-based net...   
32   B2B model for appliances and electrical products   
45  Online financial marketplace for customized ra...   
54  It connects producers of food directly to reta...   
55  It offers games that are available across Indi...   
65                  Home automation solution provider   
75  It is a sports discovery app that helps to fin...   

                                            founders  \
5                                        Nitin Gupta   
6                          Vivek AG, Veekshith C Rai   
10                           Abhay Bhat, Kinnar Shah   
14                       Ahana Gautam, Udit Kejriwal   
19                          Mohammed, Shashwat Diesh   
23                    Dhiraj Naubhar, Dheeraj Bansal   
32                                         KS Bhatia   
45                                     Rachna Suneja   
54  Thirukumaran Nagarajanin, Vasudevan Chinnathambi   
55                     Rubianca Wadhwa, Sahil Wadhwa   
65                           Dhruv Ratra, Swati Vyas   
75              Neeraj Aggarwala, Akhil Ramachandran   

                                             investor       amount  \
5                                         Undisclosed  Undisclosed   
6   Social Capital, AngelList India, Gemba Capital...  Undisclosed   
10                                  Venture Catalysts  Undisclosed   
14                                    Matrix Partners  Undisclosed   
19                          Kunal Bahl, Rohit Bansal.  Undisclosed   
23                          Inflection Point Ventures  Undisclosed   
32                                         Dinesh Dua  Undisclosed   
45                             Fintech innovation lab  Undisclosed   
54                                  Walmart, Flipkart  Undisclosed   
55                                     Sunil Gavaskar  Undisclosed   
65                                   Rockstud Capital  Undisclosed   
75  Inflection Point Ventures, SucSEED Venture Par...  Undisclosed   

           stage  funded_year  
5            NaN         2019  
6            NaN         2019  
10           NaN         2019  
14           NaN         2019  
19  Pre series A         2019  
23           NaN         2019  
32           NaN         2019  
45           NaN         2019  
54           NaN         2019  
55           NaN         2019  
65      Series B         2019  
75      Series B         2019

In [61]:
#creating a function on amount column for 2019
def clean_amount():
    # copy original amount columns
    amount_column = df_2019["amount"].copy().str.replace(",","") 
    
    # extract values in dollars($)
    amount_in_dollar = amount_column[amount_column.str.startswith("$")]
    # strip off the dollar symbol
    amount_in_dollar = amount_in_dollar.str.lstrip("$")
    # Replace the Unclean columns with the clean one

    amount_column.loc[amount_in_dollar.index] = amount_in_dollar
    
    # convert the clean column to numeric
    amount_column = pd.to_numeric(amount_column,errors="coerce")
    # update the amount column 
    df_2019["amount"] = amount_column

In [62]:
clean_amount()

In [63]:
df_2019.amount.unique()

array([6.300e+06, 1.500e+08, 2.800e+07, 3.000e+07, 6.000e+06,       nan,
       1.000e+06, 2.000e+07, 2.750e+08, 2.200e+07, 5.000e+06, 1.405e+05,
       5.400e+08, 1.500e+07, 1.827e+05, 1.200e+07, 1.100e+07, 1.550e+07,
       1.500e+06, 5.500e+06, 2.500e+06, 1.400e+05, 2.300e+08, 4.940e+07,
       3.200e+07, 2.600e+07, 1.500e+05, 4.000e+05, 2.000e+06, 1.000e+08,
       8.000e+06, 1.000e+05, 5.000e+07, 1.200e+08, 4.000e+06, 6.800e+06,
       3.600e+07, 5.700e+06, 2.500e+07, 6.000e+05, 7.000e+07, 6.000e+07,
       2.200e+05, 2.800e+06, 2.100e+06, 7.000e+06, 3.110e+08, 4.800e+06,
       6.930e+08, 3.300e+07])

### Cleaning the stage column for 2019

In [64]:
# Replacing the sector column values with the new sectors
df_2019["stage"]= df_2019["stage"].replace(modified_stage)
df_2019.stage.unique()

array([nan, 'Series C', 'Pre-Seed funding', 'Series D', 'Series A',
       'Series G', 'Series B', 'Seed funding', 'Series E', 'Series F'],
      dtype=object)

In [66]:
# call the impute_amount_column function
df_2019 = impute_amount_column(df_2019,"stage","amount")

## Data Cleaning on the 2020 Dataset

In [67]:
#First dropping unuseful columns 
df_2020.drop(columns =  "column10", inplace = True)

#Adding the year column
df_2020["funded_year"] = 2020

In [68]:
df_2020= df_2020.rename(columns=lambda x: x.lower().replace(' ', '_')) 
#dropping duplicates
df_2020.drop_duplicates(inplace=True)

In [69]:
df_2020.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1052 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   company_brand  1052 non-null   object 
 1   founded        839 non-null    float64
 2   headquarter    958 non-null    object 
 3   sector         1039 non-null   object 
 4   what_it_does   1052 non-null   object 
 5   founders       1040 non-null   object 
 6   investor       1014 non-null   object 
 7   amount         799 non-null    float64
 8   stage          590 non-null    object 
 9   funded_year    1052 non-null   int64  
dtypes: float64(2), int64(1), object(7)
memory usage: 90.4+ KB


In [70]:
df_2020.isnull().sum()

company_brand      0
founded          213
headquarter       94
sector            13
what_it_does       0
founders          12
investor          38
amount           253
stage            462
funded_year        0
dtype: int64

There is no cleaning of the amount column in string format as it is already of datatype integer. It however has 253 null values that needs to be filled with the median

In [71]:
# Replacing the sector column values with the new sectors
df_2020["stage"]= df_2020["stage"].replace(modified_stage)
df_2020.stage.unique()

array([None, 'Pre-Seed funding', 'Seed funding', 'Series A', 'Series C',
       'Series B', 'Debt', 'Series E', 'Bridge', 'Series D', 'Series F',
       'Edge', 'Series H', 'Seed Round & Series A', 'Mid series'],
      dtype=object)

In [72]:
# call the impute_amount_column function
df_2020 = impute_amount_column(df_2020,"stage","amount")

## Data Cleaning on the 2021 Dataset

In [73]:
#Cleaning the column names of the 2021_dataset
df_2021= df_2021.rename(columns=lambda x: x.lower().replace(' ', '_'))
#Adding the year column
df_2021["funded_year"] = 2021

In [74]:
df_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   company_brand  1209 non-null   object 
 1   founded        1208 non-null   float64
 2   headquarter    1208 non-null   object 
 3   sector         1209 non-null   object 
 4   what_it_does   1209 non-null   object 
 5   founders       1205 non-null   object 
 6   investor       1147 non-null   object 
 7   amount         1206 non-null   object 
 8   stage          781 non-null    object 
 9   funded_year    1209 non-null   int64  
dtypes: float64(1), int64(1), object(8)
memory usage: 94.6+ KB


### Cleaning the amount column for 2021

In [75]:
df_2021.amount.unique()

array(['$1,200,000', '$120,000,000', '$30,000,000', '$51,000,000',
       '$2,000,000', '$188,000,000', '$200,000', 'Undisclosed',
       '$1,000,000', '$3,000,000', '$100,000', '$700,000', '$9,000,000',
       '$40,000,000', '$49,000,000', '$400,000', '$300,000',
       '$25,000,000', '$160,000,000', '$150,000', '$1,800,000',
       '$5,000,000', '$850,000', '$53,000,000', '$500,000', '$1,100,000',
       '$6,000,000', '$800,000', '$10,000,000', '$21,000,000',
       '$7,500,000', '$26,000,000', '$7,400,000', '$1,500,000',
       '$600,000', '$800,000,000', '$17,000,000', '$3,500,000',
       '$15,000,000', '$215,000,000', '$2,500,000', '$350,000,000',
       '$5,500,000', '$83,000,000', '$110,000,000', '$500,000,000',
       '$65,000,000', '$150,000,000,000', '$300,000,000', '$2,200,000',
       '$35,000,000', '$140,000,000', '$4,000,000', '$13,000,000', None,
       '$Undisclosed', '$2000000', '$800000', '$6000000', '$2500000',
       '$9500000', '$13000000', '$5000000', '$8000000',

In [76]:
df_2021.drop_duplicates(inplace=True)

In [77]:
#checking for values that do not have the dollar sign attached to them for further investigation
df_2021[~df_2021.amount.str.get(0).isin(['$'])]

company_brand  founded             headquarter  \
7                 Qube Health   2016.0                  Mumbai   
8                    Vitra.ai   2020.0               Bangalore   
21                      Uable   2020.0               Bangalore   
39                   TruNativ   2019.0                  Mumbai   
54                     AntWak   2019.0               Bangalore   
64                Rage Coffee   2018.0               New Delhi   
67                      Kudos   2014.0                    Pune   
86                   Onsurity   2020.0               Bangalore   
98                    FanPlay   2020.0          Computer Games   
242        Fullife Healthcare   2009.0  Pharmaceuticals\t#REF!   
257                   MoEVing   2021.0         Gurugram\t#REF!   
316                 Hubhopper   2015.0               New Delhi   
319             Battery Smart   2019.0               New Delhi   
321                   Onelife   2019.0                  Mumbai   
325                     Aisle   2014.0               Bangalore   
334                   Zingbus   2019.0                 Haryana   
336                  Codingal   2020.0               Bangalore   
345                Hypd Store   2020.0               New Delhi   
367                 PrepBytes   2019.0                Gurugram   
376                     ExMyB   2018.0                Gurugram   
379                      Qin1   2019.0                   Noida   
383                   KIKO TV   2020.0                  Mumbai   
384               Quali55Care   2018.0                  Mumbai   
386                    RevFin   2018.0               New Delhi   
393                      Zouk   2016.0                  Mumbai   
406           Digit Insurance   2016.0               Bangalore   
429                    Bolkar   2018.0               Bangalore   
430           Wright Research   2019.0                  Mumbai   
437            Oneiric Gaming   2020.0              Chandigarh   
452                   Agri10x   2018.0                    Pune   
538               Little Leap   2020.0               New Delhi   
545                 AdmitKard   2016.0                   Noida   
551                     BHyve   2020.0                  Mumbai   
674              MYRE Capital   2020.0                  Mumbai   
677          Saarthi Pedagogy   2015.0               Ahmadabad   
724                 Instoried   2018.0               Bangalore   
732                  Homingos   2018.0               Bangalore   
733   Anar Business Community   2020.0                  Mumbai   
735                   Settlrs   2018.0               Bangalore   
766                      Wysa   2015.0               Bangalore   
769                   Firmway   2016.0                  Mumbai   
794                    Bijnis   2014.0               New Delhi   
801                   Virohan   2018.0                Gurugram   
807                 ParkSmart   2018.0               New Delhi   
810              Spark Studio   2020.0               Bangalore   
814                     Keito   2016.0                    Pune   
820                  LegalPay   2019.0               New Delhi   
824               Avalon Labs   2017.0               Bangalore   
827                   Rezo.ai   2017.0                   Noida   
833                   Polygon   2017.0                  Mumbai   
846                  Ingenium   2018.0               New Delhi   
853                   Celcius   2020.0                  Mumbai   
1100                 Sochcast   2020.0     Online Media\t#REF!   
1148                Godamwale   2016.0                  Mumbai   

                                                 sector  \
7                                            HealthTech   
8                                          Tech Startup   
21                                               EdTech   
39                                     Food & Beverages   
54                                               EdTech   
64                             

In [78]:
#creating a function to put the roaming amount values back in the right amount column
def replace_original(df, row_index, column_a, column_b):
    """Replaces the original amount value with the current wrong value if possible"""
   

    #Parameters:
    #df (DataFrame): The DataFrame containing the columns.
    #column1 (str): Name of the first column.
    #column2 (str): Name of the second column.
    #row_index (int): Index of the row where values should be swapped.

    #Returns:
    #DataFrame: The DataFrame with values swapped in the specified row.
    
    # Get the values to be swapped
    first_value = df.loc[row_index, column_a]
    second_value = df.loc[row_index, column_b]

    # Swap the values
    df.loc[row_index, column_a] = second_value
    df.loc[row_index, column_b] = first_value

    return df


In [79]:
#calling the function of detected rows
replace_original(df_2021,98,'amount','stage')
replace_original(df_2021,242,'amount','stage')
replace_original(df_2021,242,'investor','amount')
replace_original(df_2021,257,'investor','amount')
replace_original(df_2021,538,'amount','stage')
replace_original(df_2021,545,'amount','stage')
replace_original(df_2021,545,'investor','amount')
replace_original(df_2021,551,'amount','stage')
replace_original(df_2021,674,'amount','stage')
replace_original(df_2021,677,'amount','stage')
replace_original(df_2021,1100,'investor','amount')
replace_original(df_2021,1148,'amount','stage')
replace_original(df_2021,1148,'investor','amount')

company_brand  founded headquarter                 sector  \
0     Unbox Robotics   2019.0   Bangalore             AI startup   
1             upGrad   2015.0      Mumbai                 EdTech   
2        Lead School   2012.0      Mumbai                 EdTech   
3            Bizongo   2015.0      Mumbai         B2B E-commerce   
4           FypMoney   2021.0    Gurugram                FinTech   
...              ...      ...         ...                    ...   
1204        Gigforce   2019.0    Gurugram  Staffing & Recruiting   
1205          Vahdam   2015.0   New Delhi       Food & Beverages   
1206    Leap Finance   2019.0   Bangalore     Financial Services   
1207    CollegeDekho   2015.0    Gurugram                 EdTech   
1208          WeRize   2019.0   Bangalore     Financial Services   

                                           what_it_does  \
0     Unbox Robotics builds on-demand AI-driven ware...   
1        UpGrad is an online higher education platform.   
2     LEAD School offers technology based school tra...   
3     Bizongo is a business-to-business online marke...   
4     FypMoney is Digital NEO Bank for Teenagers, em...   
...                                                 ...   
1204                  A gig/on-demand staffing company.   
1205  VAHDAM is among the world’s first vertically i...   
1206  International education loans for high potenti...   
1207  Collegedekho.com is Student’s Partner, Friend ...   
1208  India’s first socially distributed full stack ...   

                                               founders  \
0                           Pramod Ghadge, Shahid Memon   
1     Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                            Smita Deorah, Sumeet Mehta   
3               Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                         Kapil Banwari   
...                                                 ...   
1204                        Chirag Mittal, Anirudh Syal   
1205                                         Bala Sarda   
1206                         Arnav Kumar, Vaibhav Singh   
1207                                       Ruchir Arora   
1208                      Vishal Chopra, Himanshu Gupta   

                                           investor        amount  \
0                       BEENEXT, Entrepreneur First    $1,200,000   
1          Unilazer Ventures, IIFL Asset Management  $120,000,000   
2                  GSV Ventures, Westbridge Capital   $30,000,000   
3                            CDC Group, IDG Capital   $51,000,000   
4     Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000   
...                                             ...           ...   
1204                                Endiya Partners      $3000000   
1205                                       IIFL AMC     $20000000   
1206                                   Owl Ventures     $55000000   
1207               Winter Capital, ETS, Man Capital     $26000000   
1208                 3one4 Capital, Kalaari Capital      $8000000   

             stage  funded_year  
0     Pre-series A         2021  
1             None         2021  
2         Series D         2021  
3         Series C         2021  
4             Seed         2021  
...            ...          ...  
1204  Pre-series A         2021  
1205      Series D         2021  
1206      Series C         2021  
1207      Series B         2021  
1208      Series A         2021  

[1190 rows x 10 columns]

In [80]:
df_2021.amount.unique()

array(['$1,200,000', '$120,000,000', '$30,000,000', '$51,000,000',
       '$2,000,000', '$188,000,000', '$200,000', 'Undisclosed',
       '$1,000,000', '$3,000,000', '$100,000', '$700,000', '$9,000,000',
       '$40,000,000', '$49,000,000', '$400,000', '$300,000',
       '$25,000,000', '$160,000,000', '$150,000', '$1,800,000',
       '$5,000,000', '$850,000', '$53,000,000', '$500,000', '$1,100,000',
       '$6,000,000', '$800,000', '$10,000,000', '$21,000,000',
       '$7,500,000', '$26,000,000', '$7,400,000', '$1,500,000',
       '$600,000', '$800,000,000', '$17,000,000', '$3,500,000',
       '$15,000,000', '$215,000,000', '$2,500,000', '$350,000,000',
       '$5,500,000', '$83,000,000', '$110,000,000', '$500,000,000',
       '$65,000,000', '$150,000,000,000', '$300,000,000', '$2,200,000',
       '$35,000,000', '$140,000,000', '$4,000,000', '$13,000,000', None,
       '$Undisclosed', '$2000000', '$800000', '$6000000', '$2500000',
       '$9500000', '$13000000', '$5000000', '$8000000',

In [81]:
# Remove the word with a value
df_2021['amount'] = df_2021['amount'].str.replace("1000000\t#REF!" , '1000000').replace('$Undisclosed', np.NaN).replace('$undisclosed',np.NaN).replace('Undisclosed', np.NaN)

In [83]:
#creating a function on amount column for 2021
def clean_amount():
    # copy original amount columns
    amount_column = df_2021["amount"].copy().str.replace(",","") 
    
    # extract values in dollars($)
    amount_in_dollar_1 = amount_column[amount_column.str.startswith("$", na=False)]
    # strip off the dollar symbol
    amount_in_dollar_1 = amount_in_dollar_1.str.lstrip("$")

    # extract values in dollars($)
    amount_in_dollar_2 = amount_column[amount_column.str.startswith("$$", na=False)]

     # strip off the dollar symbol
    amount_in_dollar_2 = amount_in_dollar_2.str.lstrip("$$")
    # Replace the Unclean columns with the clean one

    amount_column.loc[amount_in_dollar_1.index] = amount_in_dollar_1
    amount_column.loc[amount_in_dollar_2.index] = amount_in_dollar_2
    
    # convert the clean column to numeric
    amount_column = pd.to_numeric(amount_column,errors="coerce")
    # update the amount column 
    df_2021["amount"] = amount_column

In [84]:
clean_amount()

### Cleaning the stage column for 2021

In [86]:
df_2021.stage.unique()

array(['Pre-series A', None, 'Series D', 'Series C', 'Seed', 'Series B',
       'Series E', 'Pre-seed', 'Series A', 'Pre-series B', 'Debt',
       'Upsparks', 'Bridge', 'Seed+', 'Series F2', 'Series A+',
       'Series G', 'Series F', 'Series H', 'Series B3', 'PE', 'Series F1',
       'Pre-series A1', 'ah! Ventures', 'ITO Angel Network, LetsVenture',
       'Early seed', 'Series D1', 'JITO Angel Network, LetsVenture',
       'Seies A', 'Pre-series', 'Series A2', 'Series I'], dtype=object)

In [87]:
# Replacing the sector column values with the new sectors
df_2021["stage"]= df_2021["stage"].replace(modified_stage)
df_2021.stage.unique()

array(['Series A', None, 'Series D', 'Series C', 'Seed funding',
       'Series B', 'Series E', 'Pre-Seed funding', 'Debt', 'Upsparks',
       'Bridge', 'Series F', 'Series G', 'Series H', 'Private Equity',
       'Seed Funding', 'Series I'], dtype=object)

In [88]:
# call the impute_amount_column function
df_2021 = impute_amount_column(df_2021,"stage","amount")

### Concatenation of the 2019-2021 dataset

In [89]:
#concatenating the datasets into a single dataframe
df = pd.concat([df_2019,df_2020,df_2021],ignore_index=True)
df.head(2)

company_brand  founded headquarter     sector  \
0  Bombay Shaving      NaN         NaN  Ecommerce   
1       Ruangguru   2014.0      Mumbai     Edtech   

                                        what_it_does  \
0         Provides a range of male grooming products   
1  A learning platform that provides topic-based ...   

                                founders              investor       amount  \
0                     Shantanu Deshpande  Sixth Sense Ventures    6300000.0   
1  Adamas Belva Syah Devara, Iman Usman.      General Atlantic  150000000.0   

      stage  funded_year  
0       NaN         2019  
1  Series C         2019

In [90]:
# check the shape of the dataset
df.shape

(2331, 10)

In [91]:
# check the general information about the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2331 entries, 0 to 2330
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   company_brand  2331 non-null   object 
 1   founded        2088 non-null   float64
 2   headquarter    2217 non-null   object 
 3   sector         2313 non-null   object 
 4   what_it_does   2331 non-null   object 
 5   founders       2312 non-null   object 
 6   investor       2228 non-null   object 
 7   amount         2068 non-null   float64
 8   stage          1404 non-null   object 
 9   funded_year    2331 non-null   int64  
dtypes: float64(2), int64(1), object(7)
memory usage: 182.2+ KB


In [92]:
#refining the stages even more based on research about stages of funding
refined_stage = {
    'Pre-Seed funding': 'Ideation',
    'Mid series' : 'Validation',
    'Seed Round & Series A': 'Validation',
    'Seed funding': 'Validation',
    'Series A':'Early Traction',
    'Series B': 'Scaling',
    'Series C': 'Scaling',
    'Series D': 'Scaling',
    'Series E': 'Scaling',
    'Series F': 'Scaling',
    'Series G': 'Scaling',
    'Series H': 'Scaling',
    'Series I': 'Scaling',
    'Post-IPO Debt':'Exit Option' ,
    'Post-IPO Equity': 'Exit Option',
    'Secondary Market': 'Exit Option',
    'Private Equity': 'Other',
    'Corporate Round': 'Other',
    'Undisclosed' : 'Other',
    'Non-equity Assistance': 'Other',
    'Debt': 'Other',
    'Bridge': 'Other',
    'Edge': 'Other',
    'Upsparks': 'Other',
    
}

In [93]:
# Replacing the sector column values with the new sectors
df["stage"]= df["stage"].replace(refined_stage)
df.stage.unique()

array([nan, 'Scaling', 'Ideation', 'Early Traction', 'Validation', None,
       'Other', 'Seed Funding'], dtype=object)

In [94]:
df.replace({'headquarter': modified_location}, inplace=True)

In [95]:
#further finetuning location
df.headquarter.unique()

array([nan, 'Mumbai', 'Chennai', 'Hyderabad', 'Pune', 'Bangalore',
       'Noida', 'New Delhi', 'Ahmedabad', 'Gurugram', 'Chandigarh',
       'Jaipur', 'Surat', 'Rajasthan', 'Indore', 'Gurgaon', 'Belgaum',
       'Andheri', 'Kolkata', 'Tirunelveli', 'Thane', None, 'Overseas',
       'Gujarat', 'Kerala', 'Jodhpur', 'Patna', 'Trivandrum', 'Cochin',
       'Samastipur', 'Tumkur', 'Kochi', 'Ludhiana', 'Dehradun', 'Kanpur',
       'Warangal', 'Hyderebad', 'Odisha', 'Bihar', 'Panaji', 'Bhopal',
       'Coimbatore', 'Ahmadabad', 'Amaravati', 'Rajsamand', 'Ranchi',
       'Faridabad', 'Undisclosed', 'Vadodara', 'Food & Beverages',
       'Pharmaceuticals\t#REF!', 'Mohali', 'Powai', 'Ghaziabad', 'Nagpur',
       'Samsitpur', 'Lucknow', 'Silvassa', 'Roorkee', 'Ambernath',
       'Panchkula', 'Mangalore', 'Telangana', 'Bhubaneswar', 'Kottayam',
       'Satara', 'Bhilwara', 'Guwahati', 'Online Media\t#REF!',
       'Gandhinagar'], dtype=object)

In [96]:
#investigating abnormal values
df[df['headquarter'] == 'Food & Beverages']

company_brand  founded       headquarter     sector  \
1377    MasterChow   2020.0  Food & Beverages  Hauz Khas   

                             what_it_does                       founders  \
1377  A ready-to-cook Asian cuisine brand  Vidur Kataria, Sidhanth Madan   

          investor    amount       stage  funded_year  
1377  WEH Ventures  461000.0  Validation         2021

In [97]:
#investigating abnormal values
df[df['headquarter'] == 'Pharmaceuticals\t#REF!']

company_brand  founded             headquarter  \
1378  Fullife Healthcare   2009.0  Pharmaceuticals\t#REF!   

                                                 sector  what_it_does  \
1378  Primary Business is Development and Manufactur...  Varun Khanna   

                                founders investor      amount    stage  \
1378  Morgan Stanley Private Equity Asia     None  22000000.0  Scaling   

      funded_year  
1378         2021

In [98]:
#investigating abnormal values
df[df['headquarter'] == 'Online Media\t#REF!']

company_brand  founded          headquarter  \
2222      Sochcast   2020.0  Online Media\t#REF!   

                                                 sector  \
2222  Sochcast is an Audio experiences company that ...   

                                           what_it_does  \
2222  CA Harvinderjit Singh Bhatia, Garima Surana, A...   

                                     founders investor  amount stage  \
2222  Vinners, Raj Nayak, Amritaanshu Agrawal     None     NaN  None   

      funded_year  
2222         2021

In [99]:
#calling function to swap values to their original columns
replace_original(df,1377,'sector','headquarter')
replace_original(df,1378,'founders','investor')
replace_original(df,1378,'founders','what_it_does')
replace_original(df,1378,'what_it_does','sector')
replace_original(df,1378,'sector','headquarter')
replace_original(df,2222,'sector','headquarter')
replace_original(df,2222,'what_it_does','headquarter')
replace_original(df,2222,'investor','headquarter')

company_brand  founded headquarter                 sector  \
0     Bombay Shaving      NaN         NaN              Ecommerce   
1          Ruangguru   2014.0      Mumbai                 Edtech   
2           Eduisfun      NaN      Mumbai                 Edtech   
3           HomeLane   2014.0     Chennai        Interior design   
4           Nu Genes   2004.0   Hyderabad               AgriTech   
...              ...      ...         ...                    ...   
2326        Gigforce   2019.0    Gurugram  Staffing & Recruiting   
2327          Vahdam   2015.0   New Delhi       Food & Beverages   
2328    Leap Finance   2019.0   Bangalore     Financial Services   
2329    CollegeDekho   2015.0    Gurugram                 EdTech   
2330          WeRize   2019.0   Bangalore     Financial Services   

                                           what_it_does  \
0            Provides a range of male grooming products   
1     A learning platform that provides topic-based ...   
2               It aims to make learning fun via games.   
3                 Provides interior designing solutions   
4     It is a seed company engaged in production, pr...   
...                                                 ...   
2326                  A gig/on-demand staffing company.   
2327  VAHDAM is among the world’s first vertically i...   
2328  International education loans for high potenti...   
2329  Collegedekho.com is Student’s Partner, Friend ...   
2330  India’s first socially distributed full stack ...   

                                   founders  \
0                        Shantanu Deshpande   
1     Adamas Belva Syah Devara, Iman Usman.   
2                             Jatin Solanki   
3              Srikanth Iyer, Rama Harinath   
4                    Narayana Reddy Punyala   
...                                     ...   
2326            Chirag Mittal, Anirudh Syal   
2327                             Bala Sarda   
2328             Arnav Kumar, Vaibhav Singh   
2329                           Ruchir Arora   
2330          Vishal Chopra, Himanshu Gupta   

                                               investor       amount  \
0                                  Sixth Sense Ventures    6300000.0   
1                                      General Atlantic  150000000.0   
2        Deepak Parekh, Amitabh Bachchan, Piyush Pandey   28000000.0   
3     Evolvence India Fund (EIF), Pidilite Group, FJ...   30000000.0   
4              Innovation in Food and Agriculture (IFA)    6000000.0   
...                                                 ...          ...   
2326                                    Endiya Partners    3000000.0   
2327                                           IIFL AMC   20000000.0   
2328                                       Owl Ventures   55000000.0   
2329                   Winter Capital, ETS, Man Capital   26000000.0   
2330                     3one4 Capital, Kalaari Capital    8000000.0   

               stage  funded_year  
0                NaN         2019  
1            Scaling         2019  
2           Ideation         2019  
3            Scaling         2019  
4                NaN         2019  
...              ...          ...  
2326  Early Traction         2021  
2327         Scaling         2021  
2328         Scaling         2021  
2329         Scaling         2021  
2330  Early Traction         2021  

[2331 rows x 10 columns]

In [100]:
#Hauz Khas is a neighbourhood in South Dlhi
df['headquarter'] = df['headquarter'].replace('Hauz Khas','Delhi')